### ISSUE TITLE
InvalidArgument false positive: direct `$this->scope($query, ...)` call to a `#[Scope]` method has its `$query` param stripped

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1034]
**Describe the bug**

Calling one `#[Scope]`-attributed method from another **as a direct instance method** (`$this->otherScope($query, ...)`, passing the `$query` builder explicitly) emits a false-positive `InvalidArgument`. The plugin strips the leading `$query` parameter from the callee's signature — which is correct for *forwarded* scope calls (`Model::scope()` via `__callStatic`, `$builder->scope()`) but wrong for a direct instance call, where PHP invokes the real method with its real, unstripped signature.

Because the leading param is removed, every argument shifts left by one: the explicitly-passed `$query` (a `Builder`) is checked against what is actually the *second* declared parameter.

```
ERROR: InvalidArgument
Argument 1 of App\Models\Widget::hasAnyName expects list<string>,
but Illuminate\Database\Eloquent\Builder<App\Models\Widget> provided (see https://psalm.dev/004)
```

**Regression note:** with legacy `scopeXxx()` methods this never happened — the direct internal call used the prefixed name (`$this->scopeHasAnyName($query, ...)`), which the scope-param stripping (keyed on the bare scope name) never matched. The `#[Scope]` attribute makes the real method name equal the scope name, so direct `$this->` calls now collide with the strip logic.

**Code to reproduce**

```php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Attributes\Scope;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Model;

final class Widget extends Model
{
    /** @param list<string> $names */
    #[Scope]
    protected function hasAnyName(Builder $query, array $names): void
    {
        $query->whereIn('name', $names);
    }

    #[Scope]
    protected function active(Builder $query): void
    {
        // Direct instance call to a sibling #[Scope] method, passing $query explicitly.
        // Valid PHP + valid Laravel. Flagged InvalidArgument by the plugin.
        $this->hasAnyName($query, ['a', 'b']);
    }
}
```

Forwarded forms — `Widget::query()->hasAnyName(['a','b'])` and `Widget::hasAnyName(['a','b'])` — are correct and must keep the stripped (`$query`-less) signature. Only the direct instance call is misreported.

**Expected behavior**

No error. A direct instance call to a `#[Scope]`/`scopeXxx()` method must be type-checked against the method's real declared signature (including the leading `$query`).

**Root cause**

`ModelMethodHandler::getMethodParams()` routes the call through `isUnresolvedBuilderMethod()` → `BuilderScopeHandler::hasScopeMethod()` (true for any `#[Scope]` method) and returns `BuilderScopeHandler::getScopeParams()`, i.e. the declared params **minus the leading `$query`** (`array_slice(..., 1)`). This stripped list is applied unconditionally — including to direct instance calls where the caller already passed `$query`.

`MethodParamsProviderEvent` exposes no AST node and no instance-vs-static flag, so the call form cannot be read directly. The usable signal is `getCallArgs()`: a *forwarded* call omits `$query`, a *direct* call supplies it (full arg count, with a `Builder`-typed expression at arg 0). Gating the strip on that — or declining (`return null`, letting Psalm resolve the real method) when arg 0 is `Builder`-assignable and the arg count matches the unstripped signature — fixes it without regressing the forwarded forms.

**Environment**

- psalm-plugin-laravel: `dev-master` (`692ea2d`)
- Laravel 12, PHP 8.4

I'll open a PR.
